src/hotspot/share/gc/g1/g1MonitoringSupport.hpp

```diff
@@ -239,11 +239,32 @@ class G1MonitoringSupport : public CHeapObj<mtGC> {
 // Scope object for java.lang.management support.
 class G1MonitoringScope : public StackObj {
   G1MonitoringSupport* _monitoring_support;
+public:
+  G1MonitoringScope(G1MonitoringSupport* monitoring_support);
+  ~G1MonitoringScope();
+};
+
+class G1IncMonitoringScope : public G1MonitoringScope {
```

Comment:
```suggestion
class G1YoungGCMonitoringScope : public G1MonitoringScope {
```
Everything about young collection is prefixed with "YoungGC", let's keep naming consistent.